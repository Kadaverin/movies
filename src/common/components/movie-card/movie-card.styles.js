import { makeStyles } from '@material-ui/core/styles';

export const useCardStyles = makeStyles({
  card: {
    maxWidth: 200,
    height: '100%',
  },

  cardActionContent: {
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'flex-start',
    justifyContent: 'space-between',
  },
});
