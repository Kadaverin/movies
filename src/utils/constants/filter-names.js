export const MOVIES_FILTER_NAMES = {
  YEAR: 'year',
  GENRES: 'genres',
};
