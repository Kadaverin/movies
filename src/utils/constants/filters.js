export const MOVIES_FILTER_NAMES = {
  YEAR: 'year',
  GENRES: 'genres',
};

export const EMPTY_YEAR_FILTER_VAL = -1;
