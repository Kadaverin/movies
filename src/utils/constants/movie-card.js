export const DEFAULT_IMG_HEIGHT = 300;
