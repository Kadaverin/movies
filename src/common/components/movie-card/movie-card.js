import React from 'react';
import { string, number } from 'prop-types';
import {
  Card,
  CardActions,
  CardActionArea,
  CardContent,
  CardMedia,
  Typography,
} from '@material-ui/core';

import Raiting from '../raiting';
import { useCardStyles } from './movie-card.styles';
import { DEFAULT_IMG_HEIGHT } from '../../../utils/constants/movie-card';

function MovieCard({
  imgUrl,
  imgHeight,
  title,
  releaseDate,
  voteCount,
  voteAverage,
}) {
  const classes = useCardStyles();

  return (
    <Card className={classes.card}>
      <CardActionArea className={classes.cardActionContent}>
        <div>
          <CardMedia
            className={classes.media}
            component="img"
            alt="poster"
            height={imgHeight}
            image={imgUrl}
          />
          <CardContent>
            <Typography gutterBottom variant="h5" component="h2">
              {title}
            </Typography>

            <Typography variant="body2" color="textSecondary" component="p">
              {releaseDate}
            </Typography>
          </CardContent>
        </div>

        <CardActions disableSpacing>
          <Raiting voteAverage={voteAverage} voteCount={voteCount} />
        </CardActions>
      </CardActionArea>
    </Card>
  );
}

MovieCard.propTypes = {
  imgUrl: string.isRequired,
  title: string.isRequired,
  releaseDate: string.isRequired,
  voteCount: number.isRequired,
  voteAverage: number.isRequired,
  imgHeight: number,
};

MovieCard.defaultProps = {
  imgHeight: DEFAULT_IMG_HEIGHT,
};

export default MovieCard;
