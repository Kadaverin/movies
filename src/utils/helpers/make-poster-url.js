import { IMAGE_URLS } from '../constants/img-urls';
import { TMDB_BASE_POSTERS_PATH } from '../configs';

export const makePosterUrl = tmdbPosterPath =>
  tmdbPosterPath && typeof tmdbPosterPath === 'string'
    ? `${TMDB_BASE_POSTERS_PATH}/${tmdbPosterPath}`
    : IMAGE_URLS.DEFAULT_POSTER;
