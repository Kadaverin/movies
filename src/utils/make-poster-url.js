import { IMAGE_URLS } from './constants/img-urls';

export const makePosterUrl = tmdbPosterPath =>
  tmdbPosterPath
    ? `https://image.tmdb.org/t/p/w500/${tmdbPosterPath}`
    : IMAGE_URLS.DEFAULT_POSTER;
