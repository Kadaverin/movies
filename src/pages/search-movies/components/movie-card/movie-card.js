import React from 'react';
import { string, number } from 'prop-types';

import {
  CardActions,
  Card,
  CardActionArea,
  CardContent,
  CardMedia,
  Typography,
} from '@material-ui/core';
import { useCardStyles } from './movie-card.styles';

function MovieCard({
  imgUrl,
  imgHeight,
  title,
  releaseDate,
  voteCount,
  voteAverage,
}) {
  const classes = useCardStyles();

  return (
    <Card className={classes.card}>
      <CardActionArea className={classes.cardActionContent}>
        <div>
          <CardMedia
            className={classes.media}
            component="img"
            alt="poster"
            height={imgHeight}
            image={imgUrl}
          />
          <CardContent>
            <Typography gutterBottom variant="h5" component="h2">
              {title}
            </Typography>

            <Typography variant="body2" color="textSecondary" component="p">
              {releaseDate}
            </Typography>
          </CardContent>
        </div>

        <CardActions disableSpacing>
          <Typography variant="body2" color="secondary" component="p">
            {voteAverage}{' '}
            <Typography variant="body2" color="textSecondary" component="span">
              ({voteCount})
            </Typography>
          </Typography>
        </CardActions>
      </CardActionArea>
    </Card>
  );
}

MovieCard.propTypes = {
  imgUrl: string.isRequired,
  title: string.isRequired,
  releaseDate: string.isRequired,
  voteCount: number.isRequired,
  voteAverage: number.isRequired,
  imgHeight: number,
};

MovieCard.defaultProps = {
  imgHeight: 300,
};

export default MovieCard;
