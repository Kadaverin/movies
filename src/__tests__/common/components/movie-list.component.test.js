import React from 'react';
import { shallow } from 'enzyme';
import { OrderedSet, Map } from 'immutable';
import MovieCard from '../../../common/components/movie-card';
import MoviesList from '../../../common/components/movies-list';
import { makePosterUrl } from '../../../utils/helpers/make-poster-url';
import { DEFAULT_IMG_HEIGHT } from '../../../utils/constants/movie-card';

const movieShapes = [
  {
    id: 1,
    poster_path: 'url1',
    title: 'card1',
    release_date: '2019-07-19',
    vote_count: 100,
    vote_average: 9.2,
  },
  {
    id: 2,
    poster_path: 'url2',
    title: 'car21',
    release_date: '2019-07-19',
    vote_count: 1000,
    vote_average: 100,
  },
];

const movies = OrderedSet(movieShapes.map(Map));

describe('common component', () => {
  describe('MovieList', () => {
    test('matches snapshot', () => {
      const list = shallow(<MoviesList movies={movies} />);

      expect(list).toMatchSnapshot();
    });

    test('maps movies to MovieCard', () => {
      const list = shallow(<MoviesList movies={movies} />);

      const renderedMovies = list.find(MovieCard);

      expect(renderedMovies.length).toEqual(movies.size);

      movieShapes.forEach((movie, index) => {
        expect(renderedMovies.at(index).props()).toEqual({
          imgUrl: makePosterUrl(movie.poster_path),
          title: movie.title,
          releaseDate: movie.release_date,
          voteCount: movie.vote_count,
          voteAverage: movie.vote_average,
          imgHeight: DEFAULT_IMG_HEIGHT,
        });
      });
    });
  });
});
