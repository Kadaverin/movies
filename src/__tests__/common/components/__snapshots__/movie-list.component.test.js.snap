// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common component MovieList matches snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-container-1"
  container={true}
  justify="space-evenly"
  spacing={2}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    key="1"
  >
    <Link
      to="/movies/1"
    >
      <MovieCard
        imgHeight={300}
        imgUrl="https://image.tmdb.org/t/p/w500/url1"
        releaseDate="2019-07-19"
        title="card1"
        voteAverage={9.2}
        voteCount={100}
      />
    </Link>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    key="2"
  >
    <Link
      to="/movies/2"
    >
      <MovieCard
        imgHeight={300}
        imgUrl="https://image.tmdb.org/t/p/w500/url2"
        releaseDate="2019-07-19"
        title="car21"
        voteAverage={100}
        voteCount={1000}
      />
    </Link>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
